black.py

```diff
@@ -434,12 +434,16 @@ def main(
             sources.update(
                 gen_python_files_in_dir(p, root, include_regex, exclude_regex, report)
             )
+            if len(os.listdir(p) ) == 0:
```

Comment:
This could be expensive if the directory is large. We should do this kind of check instead below where we print the message about "No paths given", so we don't incur more I/O cost in the normal case.

Reply:
Or, we could just specialize the logic below to say
```python
if len(sources) == 0:
  if verbose or not quiet:
    if len(src) == 0:
      out("No paths given. Nothing to do 😴")
    else:
      out("Paths given contain no files to be formatted. 🤷")
```

And there's no need to `listdir()` even in the error case